File: EXERCICIOS/Subsequencia.c
#include <stdio.h>


int verificar(int n1, int n2, int vet1[], int vet2[]){
int resp=0;

int percorreu=0;
for(int x=0; x<n1; x++){

    if(vet1[x]==vet2[percorreu]){
        percorreu++;
    }

}

if(percorreu==n2){
    resp=1;
}

return resp;
}

int main () {

   int vet1[100], vet2[100];
   int n1, n2;

   scanf("%d %d", &n1, &n2);

   for(int x=0; x<n1; x++){

    scanf("%d", &vet1[x]);
   }
   
   for(int x=0; x<n2; x++){

    scanf("%d", &vet2[x]);
   }


   if(verificar(n1, n2, vet1, vet2)){
    printf("S");
   }
   else{
    printf("N");
   }

    return 0;
}

File: EXERCICIOS/Subsequencia.java
import java.util.Scanner;
public class Subsequencia {
    

    public static boolean verificar(int []vet1, int []vet2, int num1, int num2){
      boolean resp=false;
      int percorreu=0;


      for(int x=0; x<num1; x++){
          
         if(vet1[x]==vet2[percorreu]){
            percorreu++;
         }

      }
      
      if(percorreu==num2){
        resp=true;
      }


       return resp;
    }

    public static void main (String args []){

    Scanner sc = new Scanner (System.in);

    int num1=sc.nextInt();
    int num2=sc.nextInt();


    int vet1[]= new int[num1];
    int vet2[]= new int[num2];

    for(int x=0; x<num1; x++){
        vet1[x]=sc.nextInt();
    }

     for(int x=0; x<num2; x++){
        vet2[x]=sc.nextInt();
    }


   if(verificar(vet1, vet2, num1, num2)){
    System.out.println("S");
   }
   else{
    System.out.println("N");
   }


    }
}
